templates/public-adventures/body.html

```diff
@@ -19,137 +19,50 @@
-                  <button class="toggle-button font-semibold rounded inline-flex justify-between 
-                     appearance-none w-full border border-gray-200 text-gray-700 p-2 rounded"
-                    onclick="$('#tags_dropdown').slideToggle('medium');">
-                    <span class="label" data-value="{{_('select_tag')}}">{{selectedTag or _('select_tag')}}</span>
-                    <svg id="keyword_arrow_tag" class="fill-current h-6 w-6" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20">
-                      <path d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z" />
-                    </svg>
-                  </button>
-                  <div class="dropdown-menu dropdown-blue z-30 mt-2 w-full" id="tags_dropdown"
-                    style="display: none; width: 100%;"
-                    _="on mutation of @style
-                        set arrow to #keyword_arrow_tag
-                        if *display == 'none'
-                            remove .rotate-180 from arrow
-                        else if not arrow.classList.contains('rotate-180')
-                            add .rotate-180 to arrow
-                        end">
-                      {% for tag in available_tags %}
-                          {% set selected = tag in selectedTag %}
-                          <div class="option {% if selected %}selected{% endif %}"
-                            {% if selected %}
-                              hx-post="/public-adventures/filter?tag={{tag}}&reset={{selectedTag}}&level={{selectedLevel}}&lang={{selectedLang}}&search={{currentSearch}}"
-                            {% else %}
-                              hx-post="/public-adventures/filter?tag={{tag + ',' +  selectedTag}}&level={{selectedLevel}}&lang={{selectedLang}}&search={{currentSearch}}"
-                            {% endif %}
-                            hx-target="#public_adventures"
-                            data-value="{{tag}}">{{tag}}</div>
-                      {% endfor %}
-                  </div>
-                </div>
+              <custom-select data-type="multiple" id="tag_select" data-label="{{ _('select_tag') }}">
+                {% for tag in available_tags %}
+                  {% set selected = tag in selectedTag %}
+                  <option {% if selected %}selected{% endif %} 
+                  {% if selected %}
+                      hx-post="/public-adventures/filter?tag={{tag}}&reset={{selectedTag}}&level={{selectedLevel}}&lang={{selectedLang}}&search={{currentSearch}}"
+                    {% else %}
+                      hx-post="/public-adventures/filter?tag={{tag + ',' +  selectedTag}}&level={{selectedLevel}}&lang={{selectedLang}}&search={{currentSearch}}"
+                    {% endif %} 
+                    hx-target="#public_adventures" value="{{tag}}">{{tag}}</option>
+                {% endfor %}
+              </custom-select>
```

Comment:
If i click on the empty option that's supposed to remove other selected options, the following happens:

![image](https://github.com/hedyorg/hedy/assets/20051470/a74be367-55ae-480f-981e-8505836af06e)

Reply:
also, i think it's best to merge this after the latest changes introduced in #5431, or at least be careful when merging them!

Reply:
> If i click on the empty option that's supposed to remove other selected options, the following happens:
> 
> ![image](https://private-user-images.githubusercontent.com/20051470/345825244-a74be367-55ae-480f-981e-8505836af06e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyMzYyMzEsIm5iZiI6MTcyMTIzNTkzMSwicGF0aCI6Ii8yMDA1MTQ3MC8zNDU4MjUyNDQtYTc0YmUzNjctNTVhZS00ODBmLTk4MWUtODUwNTgzNmFmMDZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE3VDE3MDUzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNiYzA1OGVjOTgxNDlhZWRhMTY1MjdmODE4ODRiMzMzOTAzOGQzYjE2YmE2YTc4OGIwZjMwOTE4MzQwZThhNTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lkRlsQNqdiX22Hd70n-rShIIVYgDvvuxdixV87Yu1gQ)

But that is the "Select all" option, if you click it, it selects all of the other options, and if you deselect it, it will deselect every option.